src/ecdsa/ellipticcurve.py

```diff
@@ -178,7 +185,7 @@ def double(self):
     x3 = (l * l - 2 * self.__x) % p
     y3 = (l * (self.__x - x3) - self.__y) % p
 
-    return Point(self.__curve, x3, y3)
+    return Point(self.__curve, x3, y3, self.__order)
```

Comment:
the order is not passed here intentionally as it makes the point creation very slow (as the order is then validated)

same in __add__ and __neg__